airbase/fetch.py

```diff
@@ -126,10 +127,11 @@ async def fetch(url: str) -> str:
                     return text
 
         async def download(url: str, path: Path) -> Path:
-            text = await fetch(url)
-            async with aiofiles.open(str(path), mode="w") as f:
-                await f.write(text)
-            return path
+            async with semaphore_files:
+                text = await fetch(url)
+                async with aiofiles.open(str(path), mode="w") as f:
+                    await f.write(text)
+                return path
```

Comment:
should the semaphore be shared for download and write?
that way the download and write operation counts as one

Reply:
I tried using the same semaphore for both, which gave me a deadlock. It might be possible to use only one semaphore, but this would require a larger rewrite, e.g. have the inner fetch without a semaphore.

But the semaphores are used to limit different resources. The `semaphore` limits remote-sever resources/connections, while the `semaphore_files` limits local filehandle/memory resources.